specs/CustomDownload.md

```diff
@@ -0,0 +1,528 @@
+                        L"Select `Cancel` to cancel the download.";
+
+                    std::wstring description = std::wstring(L"URI: ") + uri.get() + L"\r\n" +
+                                               L"Mime type: " + mimeType.get() + L"\r\n" +
+                                               L"Download size in bytes: " + std::to_wstring(downloadSizeInBytes) +
+                                               L"\r\n";
+
+                    TextInputDialog dialog(
+                        m_appWindow->GetMainWindow(), L"Download Starting", prompt.c_str(),
+                        description.c_str(), resultFilePath.get());
+                    if (dialog.confirmed)
+                    {
+                        // If user selects `OK`, the download will complete normally.
+                        // Result file path will be updated if a new one was provided.
+                        CHECK_FAILURE(args->put_ResultFilePath(dialog.input.c_str()));
+                    }
+                    else
+                    {
+                        // If user selects `Cancel`, the download will be canceled.
+                        CHECK_FAILURE(args->put_Cancel(TRUE));
+                    }
+                };
+
+                wil::com_ptr<ICoreWebView2Deferral> deferral;
+                CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                // This function can be called to show the download dialog and
+                // complete the event at a later time, allowing the developer to
+                // perform async work before the event completes.
+                m_completeDeferredDownloadEvent = [showDialog, deferral] {
+                    showDialog();
+                    CHECK_FAILURE(deferral->Complete());
+                };
+
+                return S_OK;
+            })
+            .Get(),
+        &m_DownloadStartingToken));
+```
+## .NET/ WinRT
+```c#
+webView.CoreWebView2.DownloadStarting += delegate (object sender, CoreWebView2DownloadStartingEventArgs args)
+{
+    CoreWebView2Deferral deferral = args.GetDeferral();
+    System.Threading.SynchronizationContext.Current.Post((_) =>
+    {
+        using (deferral)
+        {
+        
+            if (_coreWebView2Settings == null)
+            {
+                _coreWebView2Settings = webView.CoreWebView2.Settings;
+            }
+            _coreWebView2Settings.ShouldDisplayDefaultDownloadDialog = false;
+
+            var dialog = new TextInputDialog(
+                title: "Download Starting",
+                description: "Enter new result file path or select OK to keep default path. Select cancel to cancel the download.",
+                defaultInput: args.ResultFilePath);
+            if (dialog.ShowDialog() == true)
+            {
+            args.ResultFilePath = dialog.Input.Text;
+            }
+            else
+            {
+            args.Cancel = true;
+            }
+        }
+    }, null);
+};
+```
```

Comment:
We're missing a sample for the new functionality / scenario covered by the doc - creating a new download UI. This could just be in C# / WinRT. No need to be fully working but if you have the IDL compiling it does make it easier to write the code in VS and ensure the code should be able to build.